crates/bdk/src/wallet/coin_selection.rs

```diff
@@ -313,7 +313,7 @@ impl CoinSelectionAlgorithm for OldestFirstCoinSelection {
 pub fn decide_change(remaining_amount: u64, fee_rate: FeeRate, drain_script: &Script) -> Excess {
     // drain_output_len = size(len(script_pubkey)) + len(script_pubkey) + size(output_value)
     let drain_output_len = serialize(drain_script).len() + 8usize;
-    let change_fee = fee_rate.fee_vb(drain_output_len);
+    let change_fee = (fee_rate * Weight::from_vb_unchecked(drain_output_len as u64)).to_sat();
```

Comment:
I agree this is a concern. My opinion is that an insanely sized `drain_script` would overflow the block size before overflowing a `u64`, so this check should ideally occur during tx building and throw an error if the drain script exceeds either `MAX_STANDARD_TX_WEIGHT` (400kwu) or more generously `MAX_BLOCK_WEIGHT` (4Mwu). Note that if the drain script is derived from a typical bitcoin address, then this should never happen. I think adding these sanity checks early in tx building would make the code more robust, instead of cramming the logic into this particular line. What would be your take on that?

I would apply the same reasoning to `size_of_change` in the constructor of `BranchAndBoundCoinSelection`, but from what I hear, the coin select module is soon to be replaced so I don't know how relevant this is now.

In any case, your suggestion is sound, so we can either fixup this line of code, or I can add a note in the PR description about adding these extra consensus/standardness checks as a followup.

Reply:
I may be overthinking. We can do both - check for overflow here, and make a note to check the drain script size elsewhere

Reply:
I don't think we need to check for overflow, in an error handling situation.
Just an `.expect()` with a nice error message would be sufficient in my opinion.
It will give enough context for the user of the source of the error.